### PR TITLE
publish: diving into jtag part 3

### PR DESCRIPTION
**Pull request by gminn**
### Summary

This is part 3 of the JTAG series! 🎉

A few tweaks for publishing:

- Add all images to `img/`
- Beef up excerpt
- Request suggestions at the end instead of beginning
- Minor formatting tweaks (including line wrapping
with `npx prettier --prose-wrap always --write`, we should
run this precommit!)
- Minor grammar tweaks

### Test Plan

Visual check.

### CODE REVIEW

**Comment by gminn** [comment]
* **#445** <a href="https://app.graphite.dev/github/pr/memfault/interrupt/445?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gminn and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by gminn** [comment]
@Zamuhrishka Excited to put up the third part of your JTAG series! Could you take a quick look, and let me know if the article looks good from your end?

File: _posts/2024-03-22-diving-into-jtag-part-3.md [hunk 1]
```diff
@@ -0,0 +1,366 @@
+---
```
Comment (gminn) [inline _posts/2024-03-22-diving-into-jtag-part-3.md:1]: I should have put the line wrapping in a separate commit so changes would be easier to review. A summary of my changes is in the PR description.

**Review by bahildebrand — APPROVED** [review]
LGTM

File: _posts/2024-03-22-diving-into-jtag-part-3.md [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+   there is an incorrect signal on pin 2 of chip `D2`, so the connection `D1:7`
+   <--> `D2:2` has some problems.
+
+## Conclusion
+
+As Socrates said, "The more I know, the more I realize I know nothing." Indeed,
+this is a challenging topic that has made me realize there is so much more
+complexity to JTAG than meets the eye! Any suggestions on things I may have
+overlooked are most welcome. Engage with me in the comments -- I look forward to
+hearing from you!
```
[the remark this answers is not in the document]
Comment (Zamuhrishka) [reply _posts/2024-03-22-diving-into-jtag-part-3.md:337]: Hi @gminn! Everything looks good. :)

File: _posts/2024-03-22-diving-into-jtag-part-3.md [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+the length and format of the `BSR` register. This information is supplied by the
+manufacturer in the BSDL file format for a particular chip. Here is an example
+of a truncated BSDL file for the STM32F407 microcontroller:
+
+```plaintext
+num       cell        port     function
+405       BC_1,        *,      CONTROL,
+404       BC_1,        PE2,    OUTPUT3,
+403       BC_4,        PE2,    INPUT,
+```
```
Comment (noahp) [inline _posts/2024-03-22-diving-into-jtag-part-3.md:196]: much nicer! could also have been a markdown table i think?
